### ISSUE TITLE
Attention map is useless in this code?

### ISSUE THREAD

**Issue by o0t1ng0o** [machine-perception-robotics-group/ABN_CelebA#3]
Hi, @Hiroshi-Fukui 

Thanks you for sharing this code about the multi-task!

I tried this code for 12 facial attributes. But I found that the attention map is almost useless.
Here are the attention maps which are generated by test.py.

Bald:
![000000_Bald](https://user-images.githubusercontent.com/16815652/66248349-7524e000-e758-11e9-807b-57abb19132c3.png)
Bangs:
![000000_Bangs](https://user-images.githubusercontent.com/16815652/66248350-75bd7680-e758-11e9-8da0-b7eb9cdf12fe.png)
Big Lips:
![000000_Big_Lips](https://user-images.githubusercontent.com/16815652/66248351-75bd7680-e758-11e9-886f-f5420f799d8a.png)
Big Nose:
![000000_Big_Nose](https://user-images.githubusercontent.com/16815652/66248352-76560d00-e758-11e9-92a9-a0f7f2679e62.png)
Black Hair:
![000000_Black_Hair](https://user-images.githubusercontent.com/16815652/66248353-76560d00-e758-11e9-8bbd-2a3438b3946b.png)
Blond Hair:
![000000_Blond_Hair](https://user-images.githubusercontent.com/16815652/66248354-76560d00-e758-11e9-9ac2-e77ac9cb8d1d.png)
Blurry:
![000000_Blurry](https://user-images.githubusercontent.com/16815652/66248355-76eea380-e758-11e9-95a3-7762abd9bb83.png)
Brown Hair:
![000000_Brown_Hair](https://user-images.githubusercontent.com/16815652/66248356-76eea380-e758-11e9-94e3-b41eed2a90c2.png)
Input image:
![input](https://user-images.githubusercontent.com/16815652/66248357-77873a00-e758-11e9-818a-5e45ac73290c.png)

Is there any problem in your code?
Could you please give me advice?
Many thanks!

**Comment by Hiroshi-Fukui** [comment]
Thank you for this repo!
First, let me confirm if I got it.
Did you re-train our ABN with 12 facial attributes? and did you use the chainer==2.1.0, cupy==1.0.3 and Python2?

**Comment by o0t1ng0o** [comment]
Hi, @Hiroshi-Fukui 
I directly trained ABN with 12 facial attributes. 
I used chainer == 6.4.0, cupy==6.4.0 and Python3.6.
So the model differs extremely when using different environment?

**Comment by Hiroshi-Fukui** [comment]
Yes, I only evaluated the old version of chainer , cupy, and python. sorry...

**Comment by o0t1ng0o** [comment]
@Hiroshi-Fukui  So is it possible to implement in PyTorch?

**Comment by o0t1ng0o** [comment]
Hi, @Hiroshi-Fukui 
I failed to set up the environment.
Could you please tell the version of CUDA, cuDNN, nccl, GPU driver and python?

Additionally, can I rewrite the code of multi-task learning based on your code repository (https://github.com/machine-perception-robotics-group/attention_branch_network) which is for image classification?
Is it possible to come out the meaningful attention maps for different facial attributes?

Many thanks!
I am looking forward to your feedback~

**Comment by Hiroshi-Fukui** [comment]
Sorry, I missed the new issues...

Sorry, I forget the versions of CUDA, cuDNN, nccl, GPU driver. The version of python is 2.
But, you can get our chainer environment by `docker pull fhiro0125/chainercv_07_`.

Maybe, you can implement the ABN on facial attributes recognition using pytorch. I could implement using ABN image classification model (, but I can't open to the public it on GitHub).

**Comment by o0t1ng0o** [comment]
Thanks for your reply~

I tried to pull the environment. 
When I use this command `docker pull fhiro0125/chainercv_07_`, it comes the error `invalid reference format`.
Is there any problem in this command?

**Comment by o0t1ng0o** [comment]
Hi, @Hiroshi-Fukui 
I successfully get your environment by `docker pull fhiro0125/chainercv_07_1`.
But the attention maps are still the same as the images above.
Could you please provide the pretrained model of chainer version?

**Comment by Hiroshi-Fukui** [comment]
Very sorry, I lost your message...

I modified this code after submitting the CVPR paper. I forget the modification, sorry...

Please try to add the two lines to `test.py`. I removed the normalizing process of an attention map.
```
min_att = 0
max_att = 30
...
item_attention = (item_attention - min_att) / (max_att - min_att)
```
When I add the lines, I can get the correct attention map like CVPR and arXiv papers.
(Please adjust the value of `max_att`. In CelebA, 30 is the best value.)
